UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -136,17 +114,19 @@ public abstract class Agent : MonoBehaviour
         BehaviorParameters m_PolicyFactory;
 
         /// <summary>
-        /// Agent parameters specified within the Editor via AgentEditor.
+        /// The maximum number of steps the agent takes before being done.
         /// </summary>
-        [HideInInspector] public AgentParameters agentParameters;
+        /// <remarks>
+        /// If set to 0, the agent can only be set to done programmatically (or
+        /// when the Academy is done).
+        /// If set to any positive integer, the agent will be set to done after
+        /// that many steps. Note that setting the max step to a value greater
+        /// than the academy max step value renders it useless.
+        /// </remarks>
+        [HideInInspector] public int maxStep;
```

Comment:
Why is this `[HideInInspector]`? Are you expected to set this from code, not the Editor?

Reply:
There is a custom inspector on the Agent